### PR TITLE
Fixed missing breadcrumbs on project page, #2728

### LINKED ISSUE #2728: Breadcrumbs are missing on project page
Not sure if I should label this as a bug or feature request, but this one feels more like a bug to me.

## Search terms

breadcrumbs headings modules index

## Expected Behavior

Expected page title breadcrumbs and heading to be consistent across all pages.

## Actual Behavior

Breadcrumbs are missing from the modules.html index page despite being on all other index pages (modules/*.html). Also the heading is an h2 instead of an h1.

## Steps to reproduce the bug

src/index.ts:
```ts
/**
 * @packageDocumentation
 * Example
 *
 * @document ../CHANGELOG.md
 * @document ../LICENSE.md
 * @document ../README.md
 */

/** @private */
export function helloWorld() {
  return "Hello World";
}
```

typedoc.json:
```json
{
  "entryPoints": ["./src/index.ts"],
  "out": "./docs"
}
```

docs/modules.html screenshot:
example

Compared to:

src/index.ts:
```ts
/**
 * @packageDocumentation
 * Example
 *
 * @document ../CHANGELOG.md
 * @document ../LICENSE.md
 */

/** @private */
export function helloWorld() {
  return "Hello World";
}
```

typedoc.json:
```json
{
  "entryPoints": ["./src/index.ts"],
  "out": "./docs",
  "projectDocuments": ["./README.md"]
}
```

docs/modules/index.html screenshot:
example2

## Environment

-   Typedoc version: 0.26.8
-   TypeScript version: 5.6.2
-   Node.js version: 20.11.1
-   OS: Windows 10

### PR DESCRIPTION
**Pull request by mrfigg**
Resolves #2728.

example

### CODE REVIEW

File: src/lib/output/themes/default/partials/header.tsx [hunk 1]
```diff
@@ -8,7 +8,9 @@ export const header = (context: DefaultThemeRenderContext, props: PageEvent<Refl
     const HeadingLevel = props.model.isProject() ? "h2" : "h1";
     return (
         <div class="tsd-page-title">
-            {!!props.model.parent && <ul class="tsd-breadcrumb">{context.breadcrumb(props.model)}</ul>}
+            {!!props.url && props.url !== "index.html" && props.url !== "hierarchy.html" && (
```
Comment (Gerrit0) [inline src/lib/output/themes/default/partials/header.tsx:11]: `props.url !== "index.html"` is the only branch here that's possible, there will always be a url at this point, and this helper isn't called for the hierarchy page

Comment (mrfigg) [reply src/lib/output/themes/default/partials/header.tsx:11]: Just as a test I added the following code before the return and it does trigger, so `header.tsx` is used for `hierarchy.html`.

```ts
if (props.url === "hierarchy.html") {
    console.log("hierarchy matched");
}
```

I do agree that the `!!props.url && ` is a bit useless on a second glance though.

Comment (Gerrit0) [reply src/lib/output/themes/default/partials/header.tsx]: Huh, I apparently got very confused when looking at this, LGTM!

**Comment by Gerrit0** [comment]
Thanks!

**Pull request merged**